diff -up mariadb-10.2.6/scripts/galera_new_cluster.sh.p42 mariadb-10.2.6/scripts/galera_new_cluster.sh
--- mariadb-10.2.6/scripts/galera_new_cluster.sh.p42	2017-06-18 22:08:10.921574458 +0200
+++ mariadb-10.2.6/scripts/galera_new_cluster.sh	2017-06-18 22:13:58.836159661 +0200
@@ -24,7 +24,13 @@ fi
 VERSION="@VERSION@@MYSQL_SERVER_SUFFIX@"
 COMPILATION_COMMENT="@COMPILATION_COMMENT@"
 
-systemctl set-environment _WSREP_NEW_CLUSTER='--wsrep-new-cluster' && \
-    systemctl start ${1:-@DAEMON_NAME@}
+echo "_WSREP_NEW_CLUSTER='--wsrep-new-cluster'" >> @sysconfdir@/sysconfig/mariadb
+ 
+service @DAEMON_NAME@ start
+
+EXIT_CODE=$?
+
+sed -i -r "/^_WSREP_NEW_CLUSTER='--wsrep-new-cluster'$/d" @sysconfdir@/sysconfig/mariadb
+
+exit $EXIT_CODE
 
-systemctl set-environment _WSREP_NEW_CLUSTER=''
