diff -up mariadb-10.1.11/scripts/galera_new_cluster.sh.servicename mariadb-10.1.11/scripts/galera_new_cluster.sh
--- mariadb-10.1.11/scripts/galera_new_cluster.sh.servicename	2016-02-11 13:57:39.739360761 +0100
+++ mariadb-10.1.11/scripts/galera_new_cluster.sh	2016-02-11 13:57:56.017395496 +0100
@@ -9,6 +9,6 @@ VERSION="@VERSION@@MYSQL_SERVER_SUFFIX@"
 COMPILATION_COMMENT="@COMPILATION_COMMENT@"
 
 systemctl set-environment _WSREP_NEW_CLUSTER='--wsrep-new-cluster' && \
-    systemctl start ${1:-mariadb}
+    systemctl start ${1:-@DAEMON_NAME@}
 
 systemctl set-environment _WSREP_NEW_CLUSTER=''
